packages/runtime-core/src/apiCreateComponent.ts

```diff
@@ -7,14 +7,19 @@ import {
 } from './apiOptions'
 import { SetupContext, RenderFunction } from './component'
 import { ComponentPublicInstance } from './componentProxy'
-import { ExtractPropTypes } from './componentProps'
+import { ExtractPropTypes, PropOptions } from './componentProps'
 import { isFunction } from '@vue/shared'
 
 // overload 1: direct setup function
 // (uses user defined props interface)
 export function createComponent<Props, RawBindings = object>(
   setup: (props: Props, ctx: SetupContext) => RawBindings | RenderFunction
 ): {
+  props: {
```

Comment:
The test cases already work without this.

Reply:
I understood now, we need `h` to test it. But how to expect a TS error? e.g. I want to test if `h(Comp, { a: {} })` won't be compiled correctly (`a` is of type `string` for example).